mlflow/models/model.py

```diff
@@ -22,18 +22,18 @@ class Model(object):
     """
 
     def __init__(self, artifact_path=None, run_id=None, utc_time_created=None, flavors=None,
-                 signature: ModelSignature = None, input_example: ModelInputExample = None,
+                 signature: ModelSignature=None, input_example: Dict[str, Any]=None,
                  **kwargs):
         # store model id instead of run_id and path to avoid confusion when model gets exported
         if run_id:
             self.run_id = run_id
             self.artifact_path = artifact_path
         self.utc_time_created = str(utc_time_created or datetime.utcnow())
         self.flavors = flavors if flavors is not None else {}
-        if signature is not None:
-            self.signature = signature
-        if input_example is not None:
-            self.input_example = input_example
+        self._signature = None
```

Comment:
Hm don't we want to assign `self._signature` to `signature` here?

Reply:
yeah, this is confusing. I wanted to use setter but the linter was complaining that _signature was not defined in the __init__ so I just did declare it as None before assigning the user value vis the setter. I will either assign directly to _signature here (the setter does not really do anything else right now) or remove the _signature assignment and disable the warning.